skbeam/core/accumulators/binned_statistic.py

```diff
@@ -0,0 +1,362 @@
+        xc,yc: int, optional
+            location (in pixels) of origin (default: image center).
+            see "cartesian" parameter for definition of x/y.
+        rrange: (float, float), optional
+            The lower and upper radial range of the bins, in pixels.
+            If not provided, all pixel r values are included.
+        phirange: (float, float), optional
+            phi range to include.  Values are in the range
+            (-180,180) degrees (default: no limits).  Phi is
+            computed as arctan(y/x) where the meaning of y and x
+            is determined by the setting of the "cartesian" parameter.
+        mask: 2-dimensional np.ndarray, optional
+            array of zero/non-zero values, same shape as image used
+            in __call__.  zero values will be ignored.
+        cartesian: bool, optional
+            if True, use "cartesian" ordering, with x corresponding
+            to matrix columns and y corresponding to matrix rows.
+            Otherwise the opposite ("matrix" ordering). (default: True).
+        """
+
+        if not cartesian:
+            # switch from matrix to cartesian by swapping axes
+            xc, yc = yc, xc
+            xsize, ysize = ysize, xsize
+
+        xc = xsize//2 if xc is None else xc
+        yc = ysize//2 if yc is None else yc
+        x = np.arange(xsize)-xc
+        y = np.arange(ysize)-yc
+        xgrid, ygrid = np.meshgrid(x, y)  # "cartesian"
+        self.expected_shape = xgrid.shape
+
+        rpix = np.sqrt(xgrid**2 + ygrid**2)
+
+        if phirange is not None:
+            if cartesian:
+                phipix = np.arctan2(ygrid, xgrid) * 180 / np.pi
+            else:
+                phipix = np.arctan2(xgrid, ygrid) * 180 / np.pi
+
+        # exclude pixels this is arguably not-ideal, forcing masked
+        # pixels to be outside the histogram range, but it's more
+        # performant, because we don't have to iterate over the array
+        # a second time on each __call__ picking out the same unmasked
+        # pixels
+
+        if rrange is None:
+            rrange = (rpix.min(), rpix.max())
+        excludeval = rrange[0]-1
+        if phirange is not None:
+            rpix[phipix < phirange[0]] = excludeval
+            rpix[phipix > phirange[1]] = excludeval
+        if mask is not None:
+            if mask.shape != self.expected_shape:
+                raise ValueError('"mask" has incorrect shape')
+            rpix[mask == 0] = excludeval
+
+        super(RadialBinnedStatistic, self).__init__(rpix.reshape(-1),
+                                                    statistic,
+                                                    bins=bins, range=rrange)
+
+    def __call__(self, values):
+        # check for what I believe could be a common error
+        if values.shape != self.expected_shape:
+            raise ValueError('"values" has incorrect shape')
+        return super(RadialBinnedStatistic, self).__call__(values.reshape(-1))
```

Comment:
I was going to suggest `ravel` but then read the docs just to be sure and it suggests this method http://docs.scipy.org/doc/numpy-1.10.4/reference/generated/numpy.ravel.html